test/src/test/java/hudson/model/ProjectTest.java

```diff
@@ -568,7 +571,12 @@ public void testDoDoDelete() throws Exception{
         auth.add(Jenkins.READ, user.getId());
         auth.add(Job.READ, user.getId());
         auth.add(Job.DELETE, user.getId());
-        List<HtmlForm> forms = j.createWebClient().login(user.getId(), "password").goTo(project.getUrl() + "delete").getForms();
+
+        JenkinsRule.WebClient wc = j.createWebClient();
+        wc.withBasicCredentials(user.getId(), "password");
+        HtmlPage p = wc.goTo(project.getUrl() + "delete");
```

Comment:
Note that use of BASIC credentials here is less realistic than the original version, since this is testing HTML UI that would in practice always be preceded by a cookie-based login. For purposes of this test it probably does not matter and the speedup may be worth the slight loss of fidelity.